swap/src/bitcoin/wallet.rs

```diff
@@ -273,7 +273,11 @@ impl WaitForTransactionFinality for Wallet {
             tracing::debug!("tx_block_height: {:?}", tx_block_height);
             let block_height = self.get_block_height().await?;
             tracing::debug!("latest_block_height: {:?}", block_height);
-            if let Some(confirmations) = block_height.checked_sub(tx_block_height) {
+            if let Some(confirmations) = block_height.checked_sub(
+                tx_block_height
+                    .checked_sub(BlockHeight::new(1))
+                    .expect("transaction must be included in block with height >= 1"),
```

Comment:
Not sure if it is more _elegant_ but it you chose to panic anyway, you could also use the non-checked API with `-`.

Reply:
I was more thinking in terms of changing the logic/naming on a larger scale to avoid this calculation. I like the `checked_sub` because it provides more context on what is going on with this calculation. Does not add anything to the runtime constraints though :)